rust/catalyst-types/src/uuid/mod.rs

```diff
@@ -32,16 +33,30 @@ pub enum CborUuidError {
     },
 }
 
-/// Decode `CBOR` encoded `UUID`.
-pub(crate) fn decode_cbor_uuid(val: &coset::cbor::Value) -> Result<uuid::Uuid, CborUuidError> {
-    let Some((UUID_CBOR_TAG, coset::cbor::Value::Bytes(bytes))) = val.as_tag() else {
-        return Err(CborUuidError::InvalidCborType);
-    };
-    let uuid = uuid::Uuid::from_bytes(
-        bytes
-            .clone()
-            .try_into()
-            .map_err(|_| CborUuidError::InvalidByteSize)?,
-    );
+/// Decode from `CBOR` into `UUID`
+pub(crate) fn decode_cbor_uuid(
```

Comment:
This should implement the minicbor `Decode` trait.
See: https://docs.rs/minicbor/0.25.1/minicbor/decode/trait.Decode.html